src/libstd/thread_local/mod.rs

```diff
@@ -205,6 +205,7 @@ macro_rules! __thread_local_inner {
 
         #[cfg(any(not(any(target_os = "macos", target_os = "linux")), target_arch = "aarch64"))]
         const _INIT: ::std::thread_local::__impl::KeyInner<$t> = {
+            #[allow(unsafe_code)]
             unsafe extern fn __destroy(ptr: *mut u8) {
                 ::std::thread_local::__impl::destroy_value::<$t>(ptr);
```

Comment:
This is one of the weird things I might have cleaned up in my post-const-fn branch - `thread_local` definitely took the longest out of all of libstd.

Would using `::std::thread_local::__impl::destroy_value::<$t>` instead of `__destroy` work at all?
Is it perhaps an ABI mismatch? If it is, can the `destroy_value` definition be changed, instead?

Also, take great care in keeping consistency across the supported combinations (IMO they should go in separate files to avoid `#[cfg]` confusion).